src/EventManagement.Services/PowerOffice/PowerOfficeService.cs

```diff
@@ -0,0 +1,87 @@
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using GoApi;
+using GoApi.Invoices;
+using GoApi.Common;
+using GoApi.Core;
+
+using losol.EventManagement.Domain;
+using System.Linq;
+using GoApi.Party;
+
+namespace losol.EventManagement.Services.PowerOffice
+{
+    public class PowerOfficeService : IPowerOfficeService
+    {
+        private readonly Go api;
+        public PowerOfficeService(IOptions<PowerOfficeOptions> options)
+        {
+            GoApi.Global.Settings.Mode = options.Value.Mode;
+            var authorizationSettings = new AuthorizationSettings
+            {
+                ApplicationKey = options.Value.ApplicationKey,
+                ClientKey = options.Value.ClientKey,
+                TokenStore = new BasicTokenStore(options.Value.TokenStoreName)
+            };
+            var authorization = new Authorization(authorizationSettings);
+            api = new Go(authorization);
+        }
+
+        public async Task CreateInvoiceAsync(Order order)
+        {
+            var customer = await createCustomerIfNotExists(order);
+            var invoice = new OutgoingInvoice 
+            {    
+                Status = OutgoingInvoiceStatus.Approved, // DOES NOT ACTUALLY SEND IT
+                OrderDate = order.OrderTime,
+                CustomerReference = order.CustomerInvoiceReference,
+                CustomerCode = customer.Code,
+                
+            };
+            foreach(var orderline in order.OrderLines)
+            {
+                var invoiceLine = new OutgoingInvoiceLine
+                {
+                    LineType = VoucherLineType.Normal,
+                    ProductCode = $"{orderline.ProductId}-{orderline.ProductVariantId}",
```

Comment:
Should we do a check if the products exist, and eventually create it if not?

Reply:
Yes, I don't think we have a choice :D